### ISSUE TITLE
代码问题

### ISSUE THREAD

**Issue by LevinLittleDream** [zju3dv/snake#178]
作者您好
def nms_abox(self, output):
        box = output['detection'][..., :4] # 中心点检测框的四个点坐标 （3，100，4）
        score = output['detection'][..., 4]  # 中心点的得分 （3，100）
        cls = output['detection'][..., 5]  # 中心点的类别 （3，100）

        batch_size = box.size(0) # 3
        cls_num = output['act_hm'].size(1) # 8

        box_score_cls = []
        for i in range(batch_size):
            box_score_cls_ = self.nms_class_box(box[i], score[i], cls[i], cls_num)
            box_score_cls_ = [torch.cat(d, dim=0) for d in list(zip(*box_score_cls_))] # 在0维按照类的方向拼接起来
            box_score_cls.append(box_score_cls_)

        box, score, cls = list(zip(*box_score_cls)) # 【（100，4）× 3】 【（100） × 3】 【 （100）× 3】
        ind = torch.cat([torch.full([len(box[i])], i) for i in range(len(box))], dim=0) # （300） 100个0、100个1、100个2
        box = torch.cat(box, dim=0) # torch.Size([300, 4])
        score = torch.stack(score, dim=1) # torch.Size([100, 3])
        cls = torch.stack(cls, dim=1) # torch.Size([100, 3])
        
        detection = torch.cat([box, score, cls], dim=1)

        return detection, ind

这里面box的维度orch.Size([300, 4])，没法与torch.Size([100, 3])在dim=1上直接进行相连啊

**Comment by pengsida** [comment]
建议你运行代码看看。
